### PR TITLE
SATA protocol added

### PR DESCRIPTION
**Pull request by bhusang**

### CODE REVIEW

File: src/protocols/sata.stanza [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  SATA2p0  ; 3.2 Gbit/s
+  SATA3p0  ; 6.0 Gbit/s
+  SATA3p1  ; 6.0 Gbit/s
+  SATA3p2  ; 6.0 Gbit/s
+  SATA3p3  ; 6.0 Gbit/s
+  SATA3p4  ; 6.0 Gbit/s
+
+doc: \<DOC>
+@brief SATA Bundle
+The Data Bundle consists of 1 lane pairs (1 RX or TX pair)
```
Comment (callendorph) [inline src/protocols/sata.stanza]: Please use the `@member` marker to document the ports of a bundle.

File: src/protocols/sata.stanza [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+`pcb-differential-routing-structure` to create.
+@return Diff-Pair Constraint Object
+<DOC>
+public defn SATA-Constraint (
+  --
+  route-struct:DifferentialRoutingStructure
+  gen:jsl/protocols/sata/SATAGen = SATA3p0
+  ) -> DP-Constraint :
+  val [s, ml] = SATA-get-skew-loss-vals(gen)
+  DP-Constraint(skew = s, loss = ml, route-struct = route-struct)
```
Comment (callendorph) [inline src/protocols/sata.stanza:127]: Given SATA uses a `lane-pair` - you may want to construct the `DP-Constraint` and then construct a `LaneConstraint` with it and then return the `LaneConstraint` instance.

File: src/protocols/sata.stanza [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    SATA3p2 : [0.0 +/- 1.0e-12, 15.0] ; 
+    SATA3p3 : [0.0 +/- 1.0e-12, 15.0] ; 
+    SATA3p4 : [0.0 +/- 1.0e-12, 15.0] ; 
+
+doc: \<DOC>
+@brief Differential impedance specified by the SATA standard
+This is a helper function that returns the expected differential
+trace impedance.
+
+@returns Upper/lower limits for the impedance.
```
Comment (callendorph) [inline src/protocols/sata.stanza:108]: The marker is `@return`

https://jit-x.slack.com/docs/TKJEB34LS/F05RW135Y0J?focus_section_id=temp:C:NQAd079954f3db94ad68699899ec

File: src/protocols/sata.stanza (hunk 2 above)
Comment (callendorph) [reply src/protocols/sata.stanza:127]: Looking at your `connect-SATA` function that may not make as much sense though. That seems to be setup to convert a customized DP-constraint to lane constraint. 

Up to you.

Comment (bhusang) [reply src/protocols/sata.stanza:128]: Yeah, I just the same code pattern that I had observed elsewhere (shamelessly copying). Open to tweaking this in the future to make things easier/consistent.

**Review by callendorph — APPROVED** [review]
The example runs successfully :+1:

Looks good

**Pull request merged**